scripts/flatcar-tmpfiles

```diff
@@ -16,18 +16,21 @@ mkdir -p "${ROOT}/etc"
 
 # readable files
 umask 022
-if [[ ! -e "${ROOT}/etc/passwd" ]]; then
-    grep -E -e "^(${COPY_USERS}):" "${BASE}/passwd" > "${ROOT}/etc/passwd"
-fi
-if [[ ! -e "${ROOT}/etc/group" ]]; then
-    grep -E -e "^(${COPY_GROUPS}):" "${BASE}/group" > "${ROOT}/etc/group"
-fi
+# Output those lines in BASE/passwd (that are to be copied) when they are not in /etc/passwd already and append to /etc/passwd.
+# But since we don't want lines/half lines being added also be used as patterns, we copy the file first to memory
+# (if it doesn't exist we are fine with empty output as patterns which means to match empty lines which inverted is all).
+PATTERNS=$(cat "${ROOT}/etc/passwd" || true)
+grep -v -F -x -f <(echo "${PATTERNS}") <(grep -E -e "^(${COPY_USERS}):" "${BASE}/passwd") >> "${ROOT}/etc/passwd"
```

Comment:
Will this result in duplicate entries if admin changes something about the users to be copied (like changing `core` comment field from `Flatcar Admin` to `Flatcar Sysadmin`)?

Would a safer thing to do be to construct pattern to be in form of `^\(user1\|user2\|…\|userN\):.*$`, then drop the `-F` flag from grep? Not sure about `-x`, probably too, since the pattern already consumes the whole line.

If my concern above is true, then this is even more likely to happen for groups as they are more likely to be modified by adding more users to some group than the user info in passwd.

Reply:
Thanks, good idea!

Reply:
Your idea works well. The `-x` makes a large difference and needs to be there, e.g., for the case that we have an empty string as pattern.

Reply:
I updated the PR and did a small manual test, will do a proper test with the built image when ready.
Thanks again for the good review and suggestions!